tools/nebula-algorithm/src/main/scala/com/vesoft/nebula/tools/algorithm/utils/NebulaUtil.scala

```diff
@@ -0,0 +1,224 @@
+          .option(Parameters.TYPE, Type.EDGE.getType)
+          .option(Parameters.HOST_AND_PORTS, hostPorts)
+          .option(Parameters.PARTITION_NUMBER, partitionNumber)
+          .option(Parameters.SPACE_NAME, nameSpace)
+          .option(Parameters.LABEL, edge)
+          .option(Parameters.RETURN_COLS, labelAndWeight(edge))
+          .load()
+        if (!hasWeight) {
+          anotherDataset = anotherDataset.select("_srcId", "_dstId")
+        }
+        anotherDataset
+      }
+    })
+    if (sparkPartition.equals("0")) {
+      dataset
+    } else {
+      dataset.repartition(sparkPartition.toInt)
+    }
+  }
+
+  /**
+    * convert String host and ports to List[HostAndPort]
+    *
+    * @param hostPorts
+    *
+    * @return
+    */
+  def hostAndPorts(hostPorts: String): util.List[HostAndPort] = {
+    val hostAndPortList: util.List[HostAndPort] = new util.ArrayList[HostAndPort]
+    val hostAndPortArray: Array[String]         = hostPorts.split(",")
+    for (hostAndPort <- hostAndPortArray) {
+      hostAndPortList.add(HostAndPort.fromString(hostAndPort))
+    }
+    hostAndPortList
+  }
+
+  /**
+    * validate the nebula configuration
+    * if labels and weightCols must be corresponding
+    *
+    * @param hostPorts
+    * @param nameSpace
+    * @param labels
+    * @param hasWeight
+    * @param weightCols
+    *
+    * @return label and weightcol map
+    */
+  def validate(hostPorts: String,
+               nameSpace: String,
+               labels: List[String],
+               hasWeight: Boolean,
+               weightCols: List[String]): mutable.Map[String, String] = {
+    // check hostPorts
+    val hostAndPortList  = new ListBuffer[HostAndPort]
+    val hostAndPortArray = hostPorts.split(",")
+    for (hostAndPort <- hostAndPortArray) {
+      hostAndPortList.append(HostAndPort.fromString(hostAndPort))
+    }
+    var metaClient: MetaClientImpl = null
+    try {
+      metaClient = NebulaUtils.createMetaClient(hostAndPortList.toList)
+      if (metaClient == null) {
+        LOG.error(s" configuration for nebula.address=${hostPorts} cannot connect nebula service.")
+        System.exit(-1)
+      }
+    } catch {
+      case e: Exception => {
+        LOG.error(s" configuration for nebula.address=${hostPorts} is wrong.")
+        throw e
+      }
+    }
+
+    // check whether nameSpace exists
+    var exitSpace = false
+    import scala.collection.JavaConverters._
+    for (spaceNameID <- metaClient.listSpaces().asScala) {
+      if (spaceNameID.getName.equals(nameSpace)) {
+        exitSpace = true
+      }
+    }
+    if (!exitSpace) {
+      LOG.error(s" Space=${nameSpace} doesn't exist. ")
+      System.exit(-1)
+    }
+
+    // check if all edge labels exist and weight col match label
+    if (hasWeight && labels.size != weightCols.size) {
+      LOG.error(
+        s" size of configuration nebula.labels=${labels.size} do not match nebula.weightCols=${weightCols.size}. ")
+      System.exit(-1)
```

Comment:
`System.exit` will cause JVM crash

Reply:
It may be more reasonable to change sys.exit to throw exception. I'll change them.